### PR TITLE
Fix Multiple Typos [7]

### PR DESCRIPTION
**Pull request by JonHanna**

### CODE REVIEW

File: src/Common/src/System/Globalization/FormatProvider.Number.cs [hunk 1]
```diff
@@ -567,7 +567,7 @@ private unsafe static bool ParseNumber(ref char* str, NumberStyles options, ref
 
             private static bool TrailingZeros(string s, int index)
             {
-                // For compatability, we need to allow trailing zeros at the end of a number string
+                // For comparability, we need to allow trailing zeros at the end of a number string
```
Comment (stephentoub) [inline src/Common/src/System/Globalization/FormatProvider.Number.cs]: This looks wrong.

Comment (JonHanna) [inline src/Common/src/System/Globalization/FormatProvider.Number.cs]: And it is.

**Comment by stephentoub** [comment]
LGTM

**Comment by jamesqo** [comment]
@JonHanna Just curious, how exactly are you fixing all of these typos so quickly? Are you doing this by hand, or running a command-line program of some sort to automate this?

**Comment by JonHanna** [comment]
Bit of each. I'd been irritated by some typos in Linq.Expressions but mostly fixing them wouldn't be relevant to what I was doing at the time, so I went looking and found the spell-check plug-in at https://github.com/EWSoftware/VSSpellChecker/wiki which IMO has too many false warnings to have running all the time, but certainly helped me get rid of most of the typos there that would annoy me when I glanced on them. That's the one project I didn't submit a PR for though, because I did that on a branch of the branch I produced #7000 in, so it can't be submitted yet.

These were produced by a mixture of running that extension and grepping for similar cases (e.g. is "ambigous" anywhere else in the repository). It was rather monotonous, but my one-year-old had a virus and insisted on sleeping on top of me for most of the weekend, and he's too large to comfortably write or code underneath now (unlike when he was smaller, he was my pair-programmer for a lot of the linq performance changes I did last year) so it was something to do while waiting for my Fallen London candle to grow back :smile: 

(Ironically, the one batch I didn't do with a sleeping child on top of me was the last one, which I made a bit of a pig's ear out of with far too many erroneous changes, and need to redo).

**Pull request merged**